platform-channels.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: page
+title: Writing platform-specific code with platform channels
+permalink: /platform-channels/
+---
+
+This guide describes how to write custom platform-specific code. If you just
+want to use existing platform-specific plugins, please see [platform
+plugins](/platform-plugins/).
+
+* TOC
+{:toc}
+
+Flutter uses a flexible system that allows you to call platform-specific APIs
+whether available in Java code on Android, or in ObjectiveC or Swift code on
+iOS.
+
+Flutter's platform-specific API support does not rely on code generation, but
+rather on a flexible message passing style:
+
+* The Flutter portion of your app sends messages to it's *host*, the iOS or
+ Android portion of your app, over a platform channel.
+
+* The *host* listens on the platform channel, and receives the message. It then
+ calls into any number of platform-specific APIs -- using their native
+ programming language -- and sends back a response to the *client*, the Flutter
+ portion.
+
+### Architectural overview: platform channels
+
+Messages are passed between the client (UI) and host (platform) using platform
+channels as illustrated in this diagram:
+
+![Platform channels architecture](/images/PlatformChannels.png)
+
+Messages and responses are passed asynchronously, to ensure the user interface
+remains responsive.
+
+On the client side, `PlatformMessageChannel` ([API][PlatformMessageChannel])
```

Comment:
nit: below you say "the class Flutter*Channel", here you just say Platform*Channel.